clang/lib/CodeGen/CGDecl.cpp

```diff
@@ -1599,9 +1599,14 @@ CodeGenFunction::EmitAutoVarAlloca(const VarDecl &D) {
       bool IsMSCatchParam =
           D.isExceptionVariable() && getTarget().getCXXABI().isMicrosoft();
 
+      // No lifetime intrinsics on coroutine promise alloca, or middle end
+      // passes will assume promise dead after lifetime.end, leading to
+      // mis-optimization
+      bool IsCoroPromise = D.getName() == "__promise";
```

Comment:
We should avoid comparing string as much as possible. And also it is not good to insert coroutines logic to `CGDecl.cpp`. We'd either doing this in CGCoroutine or CoroEarly.